pandas/core/dtypes/missing.py

```diff
@@ -206,7 +207,18 @@ def _isna(obj):
     elif isinstance(obj, ABCSeries):
         result = _isna_array(obj._values)
         # box
-        result = obj._constructor(result, index=obj.index, name=obj.name, copy=False)
+        if isinstance(obj.dtype, ArrowDtype):
+            result = obj._constructor(
+                result,
+                index=obj.index,
+                name=obj.name,
+                copy=False,
+                dtype="bool[pyarrow]",
+            )
+        else:
+            result = obj._constructor(
+                result, index=obj.index, name=obj.name, copy=False
```

Comment:
```suggestion
                new_dtype = "bool[pyarrow]" if isinstance(obj.dtype, ArrowDtype) else None
                result = obj._constructor(result, index=obj.index, name=obj.name, copy=False, dtype=new_dtype)
```
Or something of the sort would be more readable.